File: micronaut-java-cli/src/main/resources/archetype-resources/src/test/java/ApplicationTest.java
package ${package};

import io.micronaut.configuration.picocli.PicocliRunner;
import io.micronaut.context.ApplicationContext;
import io.micronaut.context.env.Environment;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;
import org.junit.Test;

import static org.junit.Assert.assertTrue;

public class ApplicationTest {

    @Test
    public void testWithCommandLineOption() throws Exception {
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        System.setOut(new PrintStream(baos));

        try (ApplicationContext ctx = ApplicationContext.run(Environment.CLI, Environment.TEST)) {
            String[] args = new String[] { "-v" };
            PicocliRunner.run(Application.class, ctx, args);

            // java-maven-cli
            assertTrue(baos.toString().contains("Hi!"));
        }
    }
}

File: micronaut-kotlin-cli/src/main/resources/archetype-resources/src/test/kotlin/ApplicationTest.kt
package ${package}

import io.micronaut.configuration.picocli.PicocliRunner
import io.micronaut.context.ApplicationContext
import io.micronaut.context.env.Environment

import org.jetbrains.spek.api.Spek
import org.jetbrains.spek.api.dsl.describe
import org.jetbrains.spek.api.dsl.it
import org.jetbrains.spek.api.dsl.on
import org.junit.jupiter.api.Assertions.*

import java.io.ByteArrayOutputStream
import java.io.PrintStream

object ApplicationTest : Spek({

    describe("${artifactId}") {
        val ctx = ApplicationContext.run(Environment.CLI, Environment.TEST)

        on("invocation with -v") {
            val baos = ByteArrayOutputStream()
            System.setOut(PrintStream(baos))

            val args = arrayOf("-v")
            PicocliRunner.run(Application::class.java, ctx, *args)

            it("should display greeting") {
                assertTrue(baos.toString().contains("Hi!"))
            }
        }

        on("other") {
            // add more tests...
        }
    }
})
